src/LVSS.cpp

```diff
@@ -7,22 +7,99 @@ uint8_t LVSS::getNodeID() {
 }
 
 uint8_t LVSS::getNumElements() {
-    return 0;
+    return OBJECT_DICTIONARY_SIZE + 1;
 }
 
 CO_OBJ_T* LVSS::getObjectDictionary() {
-    return nullptr;
+    return &objectDictionary[0];
 }
 
-LVSS::LVSS(TPS2HB50BQ1* powerSwitches[POWER_SWITCHES_SIZE]) {
+LVSS::LVSS(TPS2HB35BQ* powerSwitches[POWER_SWITCHES_SIZE], IO::GPIO& vicorFT, ACS71240 acs71240) : vicorFT(vicorFT), boardEN({0}), acs71240(acs71240), state(State::INITIALIZATION) {
     for (int i = 0; i < POWER_SWITCHES_SIZE; i++) {
         this->powerSwitches[i] = powerSwitches[i];
     }
 }
 
-// temporarily commented out to make compiler happy
-//LVSS::LVSS(TPS2HB50BQ1 powerSwitchArr[POWER_SWITCHES_SIZE]) : PowerSwitches(powerSwitchArr), currentSensor(currentSensor) {
-//}
-//LVSS::LVSS(TPS2HB50BQ1* powerSwitches) : PowerSwitches(powerSwitches) {}
-//
+/* LVSS Vicor DCM4623 State Machine */
+void LVSS::process() {
+    switch (state) {
+    case State::INITIALIZATION:
+        initState();
+        break;
+
+    case State::IDLE:
+        idleState();
+        break;
+    }
+}
+
+void LVSS::initState() {
+    /* Check if it is a new state */
+    if (isNewState) {
+        isNewState = false;
+        log::LOGGER.log(log::Logger::LogLevel::INFO, "Entering initialization state");
+    }
+
+    VICOR_FAULT_ACTIVE_STATE = vicorFT.readPin();
+
+    if (time::millis() >= 101 && VICOR_FAULT_ACTIVE_STATE == IO::GPIO::State::LOW) {
+        state = State::IDLE;
+        isNewState = true;
+    }
+}
+
+void LVSS::idleState() {
+    /* Check if it is a new state */
+    if (isNewState) {
+        isNewState = false;
+        log::LOGGER.log(log::Logger::LogLevel::INFO, "Entering idle state");
+    }
+    /* Assigns the VCU signal to the union bit field */
+    this->boardEN.val = VCUBoardSig;
+
+    log::LOGGER.log(log::Logger::LogLevel::INFO, "Battery: %d\r\nHIB: %d\r\nTMS: %d\r\nHUDL: %d\r\nACC: %d\r\nGUB: %d\r\n", boardEN.batt, boardEN.hib, boardEN.tms, boardEN.hudl, boardEN.acc, boardEN.gub);
+
+    /* Turn on boards */
+    powerSwitches[0]->setPowerSwitchStates(boardEN.batt, boardEN.hib);// Turn on Battery and HIB
+    powerSwitches[1]->setPowerSwitchStates(boardEN.tms, boardEN.hudl);// Turn on TMS and HUDL
+    powerSwitches[2]->setPowerSwitchStates(boardEN.acc, boardEN.gub); // Turn on Acc and GUB
+
+    /* Current Checking */
+    PowerSwitchState.battCurrent = powerSwitches[0]->getCurrent(1);
+    PowerSwitchState.tmsCurrent = powerSwitches[1]->getCurrent(1);
+    PowerSwitchState.accCurrent = powerSwitches[2]->getCurrent(1);
+
+    if (PowerSwitchState.battCurrent == -1 || PowerSwitchState.tmsCurrent == -1 || PowerSwitchState.accCurrent == -1) {
+        log::LOGGER.log(log::Logger::LogLevel::INFO, "Power Switch Error\r\n");
+        switchFaultstatus = 1;
```

Comment:
This should also be a bitfield of faults to indicate which board is having what fault instead of a single value. This would probably be best as a single struct that can be initialized with a list of comparisons. Check TPDO map that we drew on the whiteboard.